architxt/database/export/relational.py

```diff
@@ -0,0 +1,328 @@
+        database_schema[group.name] = create_table_for_group(group, metadata, children)
+
+    relation = get_relation_from_forest(forest, schema)
+    for group, children in schema.relations.items():
+        if group.name in relation["1-n"]:
+            add_foreign_keys_to_table(database_schema, group, children, relation)
+        else:
+            create_table_for_relation(database_schema, children, metadata)
+
+    metadata.create_all(conn)
+    return relation
+
+
+def create_table_for_group(group: NodeLabel, metadata: MetaData, children: set[NodeLabel]) -> Table:
+    """
+    Create a table for the given group.
+
+    :param group: The group (NodeLabel) to create a table for.
+    :param metadata: SQLAlchemy metadata to attach the table to.
+    :param children: The child nodes of the group.
+    :return: SQLAlchemy Table object.
+    """
+    columns = create_all_columns(group, children)
+    return Table(group.name, metadata, *columns)
+
+
+def add_foreign_keys_to_table(
+    database_schema: dict, group: NodeLabel, children: tuple[NodeLabel, NodeLabel], relation: dict
+) -> None:
+    """
+    Add foreign key constraints to the database schema.
+
+    :param database_schema: The dictionary of tables in the database schema.
+    :param group: The group (NodeLabel) defining the relation.
+    :param children: The child nodes related to the group.
+    """
+    if relation["1-n"][group.name]["source"]:
+        source = database_schema[children[1].name.replace(" ", "")]
+        target = database_schema[children[0].name.replace(" ", "")]
+    else:
+        source = database_schema[children[0].name.replace(" ", "")]
+        target = database_schema[children[1].name.replace(" ", "")]
+
+    column_name = group.name if source.name == target.name else target.name + "ID"
+    target_column_name = target.primary_key.columns.keys()[0]
+
+    database_schema[source.name].append_column(Column(column_name, ForeignKey(f"{target.name}.{target_column_name}")))
+
+
+def create_table_for_relation(
+    database_schema: dict,
+    children: tuple[NodeLabel, NodeLabel],
+    metadata: MetaData,
+) -> None:
+    """
+    Create a table for the given relation.
+
+    :param database_schema: The dictionary of tables in the database schema.
+    :param group: The group (NodeLabel) defining the relation.
+    :param children: The child nodes related to the group.
+    """
+    group1 = database_schema[children[0].name.replace(" ", "")]
+    group2 = database_schema[children[1].name.replace(" ", "")]
+
+    group_name = group1.name + "_" + group2.name
+    database_schema[group_name] = Table(group_name, metadata)
+    database_schema[group_name].append_column(
+        Column(group1.name + "ID", ForeignKey(f"{group1.name}.Architxt_{group1.name + 'ID'}"), primary_key=True)
+    )
+    database_schema[group_name].append_column(
+        Column(group2.name + "ID", ForeignKey(f"{group2.name}.Architxt_{group2.name + 'ID'}"), primary_key=True)
```

Comment:
The prefix should be configurable. We may want to use a lowercase version as default value. Ideally we could take a factory in argument to be able to customize the coumn id name. Such as:

```python
create_table_for_relation(database, children, metadata, lambda table_name: f'architxt_{table_name}ID')
```